tensorflow/core/kernels/training_ops.cc

```diff
@@ -2615,34 +2733,26 @@ class SparseApplyFtrlOp : public OpKernel {
                 errors::InvalidArgument("indices must be one-dimensional"));
 
     const Tensor& lr = ctx->input(5);
-    OP_REQUIRES(ctx,
-                TensorShapeUtils::IsScalar(lr.shape()) &&
-                    lr.scalar<T>()() > static_cast<T>(0),
-                errors::InvalidArgument("lr is not a positive scalar: ",
+    OP_REQUIRES(ctx, TensorShapeUtils::IsScalar(lr.shape()),
```

Comment:
They were removed out of necessity, because the values cannot be accessed if they are on the GPU. Other training ops with existing GPU implementations also do not check these constraints.

Reply:
Can we make these checks run only if `Device` is `CPUDevice`?

It also wasn't clear to me that these checks are redundant -- will the op produce correct results even if they are violated?  If not, is this fact documented somewhere?